pkg/util/util.go

```diff
@@ -14,6 +14,67 @@ import (
 	kubeclientpkg "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// Util is a wrapper object for actual util functions to allow for easier mocking/testing.
+// Currently, it doesn't have all actual util functions, only mockable ones are defined
+type Util interface {
+	SetCondition(conditions *[]gcpv1alpha1.Condition, status corev1.ConditionStatus, reason string, message string) error
+}
+
+type gcpUtil struct {
+}
+
+// NewUtil returns a valid util object with bound functions that can be mockable
+func NewUtil() Util {
+	return &gcpUtil{}
+}
+
+// SetCondition sets a condition on a custom resource's status
+func (u *gcpUtil) SetCondition(conditions *[]gcpv1alpha1.Condition, status corev1.ConditionStatus, reason string, message string) error {
```

Comment:
I like that you change this to be a method of a struct instead of a utility function. However, I think it shouldn't be called Util but related to the Condition type? `ConditionManager` or something like that? If the State object was unified, I'd say make it a method of that type. But it's not, and probably better won't be.

Reply:
it's now moved to `condition/conditions.go`